Instead of generating an error saying a 2.x SDK is needed, we generate an error loading the NETBuildExtensionsError task

# Repro steps

Check out the `build_fail` branch of https://github.com/multiarc/SDKRepro/

Build with full Framework MSBuild:

`msbuild /restore`

# Expected

Error message: The version of Microsoft.NET.Sdk used by this project is insufficient to support references to libraries targeting .NET Standard 1.5 or higher.  Please install version 2.0 or higher of the .NET Core SDK.

# Actual

Error message:

> C:\Program Files (x86)\Microsoft Visual Studio\Preview\Enterprise\MSBuild\Microsoft\Microsoft.NET.Build.Extensions\Microsoft.NET.Build.Extensions.NETFramework.targets(65,5): error MSB4062: The "NETBuildExtensionsError" task could not be loaded from the assembly C:\Program Files (x86)\Microsoft Visual Studio\Preview\Enterprise\MSBuild\Microsoft\Microsoft.NET.Build.Extensions\\tools\net46\Microsoft.NET.Build.Extensions.Tasks.dll.  Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [C:\git\repro\2045-Pagination\src\A\A.csproj]
>
> C:\Program Files\dotnet\sdk\1.1.7\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.Sdk.targets(92,5): Cannot find project info for 'C:\git\repro\2045-Pagination\src\B\B.csproj'. This can indicate a missing project reference. [C:\git\repro\2045-Pagination\src\A\A.csproj]

# Root cause

Microsoft.NET.Build.Extensions.Tasks.dll no longer includes the `NETBuildExtensionsError` task.  Instead it includes the `NETSdkError` task.  This is because the `DefineConstants` property in the project is no longer applying because it was moved before the `Sdk.props` import, and there is a `DefineConstants` in repo toolset that is overriding it.

# Meme

![image](https://user-images.githubusercontent.com/145043/37548198-77a0bbe0-2933-11e8-8af4-4b48dda6ebac.png)

--- a/src/Tasks/Microsoft.NET.Build.Extensions.Tasks.UnitTests/GivenANETBuildExtensionsError.cs
+++ b/src/Tasks/Microsoft.NET.Build.Extensions.Tasks.UnitTests/GivenANETBuildExtensionsError.cs
@@ -1,0 +1,19 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+
+namespace Microsoft.NET.Build.Tasks.UnitTests
+{
+    public class GivenANETBuildExtensionsError
+    {
+        [Fact]
+        public void It_is_compiled_with_extensions_specific_name()
+        {
+            // Regression test for https://github.com/dotnet/sdk/issues/2061
+            // Infrastructure changes made #if EXTENSIONS that changes the task name to not apply.
+            // This test would fail to compile if we made that mistake again.
+            new NETBuildExtensionsError();
+        }
+    }
+}

--- a/src/Tasks/Microsoft.NET.Build.Extensions.Tasks/Microsoft.NET.Build.Extensions.Tasks.csproj
+++ b/src/Tasks/Microsoft.NET.Build.Extensions.Tasks/Microsoft.NET.Build.Extensions.Tasks.csproj
@@ -5,7 +5,6 @@
   <!-- This is a smaller build of Microsoft.NET.Build.Tasks to be used outside of SDK projects -->
 
   <PropertyGroup>
-    <DefineConstants>$(DefineConstants);EXTENSIONS</DefineConstants>
     <Description>The MSBuild targets and tasks which extend MSBuild's common targets.</Description>
     <OutputType>Library</OutputType>
     <PackageId>Microsoft.NET.Build.Extensions</PackageId>
@@ -26,6 +25,7 @@
   <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
 
   <PropertyGroup>
+    <DefineConstants>$(DefineConstants);EXTENSIONS</DefineConstants>
     <IsPackable>true</IsPackable>
     <PackageLayoutOutputPath>$(ArtifactsBinDir)Sdks\$(PackageId)\</PackageLayoutOutputPath>
   </PropertyGroup>

